### DISCUSSION TITLE
Putting @main in a mixin

### DISCUSSION THREAD

**Discussion by kincsescsaba** [lihaoyi/mainargs#144] [Q&A]
Hey there,

I'm not sure if this is an issue or more like a feature request, but when I try to place the `@main` annotation of `mainargs` in a trait, that is mixed into the class that contains the main method, I get a `No @main methods declared` error.

I tried referencing the self type from `ParserForMethods` but that did not fix the issue.

Here's how the approach looks like:
```scala
abstract class CaskMainWithArgs extends cask.Main:
  self: Conf =>
  private var args_ : Seq[String] = uninitialized
  def args: Seq[String] = args_
  
  override def main(argsArray: Array[String]): Unit =
    if Option(args_).isEmpty then
      args_ = argsArray.toSeq
      ParserForMethods(self).runOrExit(argsArray)
      super.main(argsArray)

trait Conf:
  self: CaskMainWithArgs =>

  @main
  def runConfInit(@arg(name = "conf", short = 'c', doc = "Set a custom configuration file")
                  configFile: String) =
    println("Hello from runConfInit")
```

Any ideas on how this behavior could be implemented?

**Comment by lihaoyi** [comment]
What version of MainArgs are you using? This should have been fixed in 0.7.1 I think https://github.com/com-lihaoyi/mainargs/pull/142

**Reply by kincsescsaba** [reply]
It's `0.7.1`. I'll check the linked ticket

**Reply by kincsescsaba** [reply]
This looks different; there is no compiler crash, just the message being printed when I start. Maybe the macro does not search in the file where the trait is located. These are in separate files in my case.

**Reply by lihaoyi** [reply]
Got it. Can you open an issue?

**Reply by kincsescsaba** [reply]
Sure, done
